apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-focused-launch.tsx

```diff
@@ -3,34 +3,53 @@
  */
 import * as React from 'react';
 import { registerPlugin as originalRegisterPlugin, PluginSettings } from '@wordpress/plugins';
-import { useSelect, useDispatch } from '@wordpress/data';
+import { useSelect } from '@wordpress/data';
+import { doAction, hasAction } from '@wordpress/hooks';
+import { addQueryArgs } from '@wordpress/url';
 import FocusedLaunchModal from '@automattic/launch';
 
 /**
  * Internal dependencies
  */
 import { LAUNCH_STORE } from './stores';
 
+const HOOK_OPEN_CHECKOUT_MODAL = 'a8c.wpcom-block-editor.openCheckoutModal';
+
 const registerPlugin = ( name: string, settings: Omit< PluginSettings, 'icon' > ) =>
 	// eslint-disable-next-line @typescript-eslint/no-explicit-any
 	originalRegisterPlugin( name, settings as any );
 
 registerPlugin( 'a8c-editor-editor-focused-launch', {
 	render: function LaunchSidebar() {
+		const currentSiteId = window._currentSiteId;
+
 		const isFocusedLaunchOpen = useSelect( ( select ) =>
 			select( LAUNCH_STORE ).isFocusedLaunchOpen()
 		);
 
-		const { closeFocusedLaunch } = useDispatch( LAUNCH_STORE );
-
 		if ( ! isFocusedLaunchOpen ) {
 			return null;
 		}
 
+		const handleCheckout = ( siteId = currentSiteId, isEcommerce = false ) => {
+			// open checkout modal assuming the cart is already updated
+			if ( hasAction( HOOK_OPEN_CHECKOUT_MODAL ) ) {
+				doAction( HOOK_OPEN_CHECKOUT_MODAL );
+				return;
+			}
+
+			// fallback: redirect to /checkout page
+			const checkoutUrl = addQueryArgs( `https://wordpress.com/checkout/${ siteId }`, {
+				// Redirect to My Home after checkout only if the selected plan is not eCommerce
+				...( ! isEcommerce && { redirect_to: `/home/${ siteId }` } ),
+			} );
+			window.top.location.href = checkoutUrl;
```

Comment:
Would wort refactoring this line and re-use the same code for the `redirectTo` prop passed to `< FocusedLaunchModal />` a few lines below in the same file

Reply:
The implementation in https://github.com/Automattic/wp-calypso/pull/47601/commits/50da20ece08415bef2b1bfc513be0190d7256003 is what you had in mind?

Reply:
What I had in mind was an even bigger refactor (with the main objective of hardcoding the `https://wordpress.com/` origin only once in the refactored function), but I realize now that it may be complicated given the `addQueryArgs`.

If it requires too long, we can leave it out of the scope of this PR

Reply:
I see we're always using `wordpress.com` here and I'm not sure it's entirely correct to ignore the environment of the parent window (wpcalypso/horizon/etc). I'm thinking at e2e testing now.

Another example: https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-steps/final-step/index.tsx#L166

Let's leave this in the backlog for now.

Reply:
I opened #47762 in the meantime